hail/src/main/scala/is/hail/HailContext.scala

```diff
@@ -614,12 +615,12 @@ class HailContext private(val sc: SparkContext,
 
   def readRows(
     path: String,
-    t: TStruct,
+    t: PStruct,
     codecSpec: CodecSpec,
     partFiles: Array[String],
     requestedType: TStruct
   ): ContextRDD[RVDContext, RegionValue] = {
-    val makeDec = codecSpec.buildDecoder(t.physicalType, requestedType.physicalType)
+    val makeDec = codecSpec.buildDecoder(t, requestedType)
```

Comment:
Great.  And eventually `t` will become an encoded type and the decoder will specify its physical type.

Reply:
yeah, exactly.

Reply:
this stuff is one of the corners where the type/ptype system breaks down currently. we need ETypes.